gems/aws-sdk-s3/lib/aws-sdk-s3/object_multipart_copier.rb

```diff
@@ -78,10 +78,15 @@ def copy_part_thread(queue)
       end
 
       def copy_part(part)
+        result = @client.upload_part_copy(part).copy_part_result
         {
-          etag: @client.upload_part_copy(part).copy_part_result.etag,
+          etag: result.etag,
+          checksum_crc32: result.checksum_crc32,
+          checksum_crc32c: result.checksum_crc32c,
+          checksum_sha1: result.checksum_sha1,
+          checksum_sha256: result.checksum_sha256,
           part_number: part[:part_number],
-        }
+        }.compact
```

Comment:
I believe `compact` was introduced in Ruby 2.4, and we still need to support Ruby 2.3.  

Additionally - I think this should be a bit more generic (ie, if additional checksum algorithms are added in the future), something more like: https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb#L164

I know the API doesn't provide the actual checksum here, but maybe we can check for any `checksum_*` fields that are set.

Reply:
I've updated to avoid specifying specific algorithms and compact.